examples/js/loaders/GLTFLoader.js

```diff
@@ -397,6 +404,51 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+	/**
+	 * Unlit Materials Extension (pending)
+	 *
+	 * PR: https://github.com/KhronosGroup/glTF/pull/1163
+	 */
+	function GLTFMaterialsUnlitExtension( json ) {
+
+		this.name = EXTENSIONS.KHR_MATERIALS_UNLIT;
+
+	}
+
+	GLTFMaterialsUnlitExtension.prototype.getMaterialType = function ( material ) {
+
+		return THREE.MeshBasicMaterial;
+
+	};
+
+	GLTFMaterialsUnlitExtension.prototype.extendParams = function ( materialParams, material, parser ) {
+
+		var pending = [];
+
+		var metallicRoughness = material.pbrMetallicRoughness;
+
+		materialParams.color = new THREE.Color( 1.0, 1.0, 1.0 );
+		materialParams.opacity = 1.0;
+
+		if ( Array.isArray( metallicRoughness.baseColorFactor ) ) {
```

Comment:
At this point, we don't actually know that metallicRoughness is defined, do we? The `material.pbrMetallicRoughness !== undefined` test in the calling function happens after the call to this extension. I think this would NPE in a (weird but valid) material lacking that particular subobject.=